### PR TITLE
Refactos: GraphQL Server to Remove unused RBAC's

### PR DESCRIPTION
**Pull request by RipulHandoo**
## Proposed changes

Removed deprecated and unused RBACs:

- Checked for the Deprecated/Unused RBAC in [roles.go.
](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/pkg/authorization/roles.go)
- Deleted RBAC definition for ```SyncHub```

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
This PR solve issue https://github.com/litmuschaos/litmus/issues/4120, by removing the unused RBAC's

### CODE REVIEW

**Comment by SohamRatnaparkhi** [comment]
LGTM 🚀

**Review by Saranya-jena — CHANGES_REQUESTED** [review]

File: chaoscenter/graphql/server/pkg/authorization/roles.go [hunk 1]
```diff
@@ -8,80 +8,48 @@ import (
...
-	UpdateProbe                  RoleQuery = "UpdateProbe"
-	GetProbe                     RoleQuery = "GetProbe"
-	ListProbes                   RoleQuery = "ListProbes"
-	MemberRoleOwnerString                  = string(model.MemberRoleOwner)
-	MemberRoleEditorString                 = string(model.MemberRoleEditor)
-	MemberRoleViewerString                 = string(model.MemberRoleViewer)
+	UserInfrastructureReg  RoleQuery = "userInfrastructureReg"
+	CreateChaosWorkFlow    RoleQuery = "CreateChaosWorkFlow"
+	ReRunChaosWorkFlow     RoleQuery = "ReRunChaosWorkFlow"
+	DeleteChaosWorkflow    RoleQuery = "DeleteChaosWorkflow"
```
[the remark this answers is not in the document]
Comment (RipulHandoo) [reply chaoscenter/graphql/server/pkg/authorization/roles.go:14]: Making updates from "workflow" to "experiment" in queries. Should changes extend to [roles.go](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/pkg/authorization/roles.go) 
```ReRunChaosExperiment     RoleQuery = "ReRunChaosExperiment"```
or elsewhere like in [this](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/graph/chaos_experiment.resolvers.go#L100)?
```authorization.MutationRbacRules[authorization.ReRunChaosExperiment]```

File: chaoscenter/graphql/server/pkg/authorization/roles.go [hunk 2]
```diff
@@ -8,80 +8,48 @@ import (
...
-	AddProbe                     RoleQuery = "AddProbe"
-	DeleteProbe                  RoleQuery = "DeleteProbe"
-	UpdateProbe                  RoleQuery = "UpdateProbe"
-	GetProbe                     RoleQuery = "GetProbe"
-	ListProbes                   RoleQuery = "ListProbes"
-	MemberRoleOwnerString                  = string(model.MemberRoleOwner)
-	MemberRoleEditorString                 = string(model.MemberRoleEditor)
-	MemberRoleViewerString                 = string(model.MemberRoleViewer)
+	UserInfrastructureReg  RoleQuery = "userInfrastructureReg"
+	CreateChaosWorkFlow    RoleQuery = "CreateChaosWorkFlow"
```
[the remark this answers is not in the document]
Comment (RipulHandoo) [reply chaoscenter/graphql/server/pkg/authorization/roles.go:12]: On it

File: chaoscenter/graphql/server/pkg/authorization/roles.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Saranya-jena) [reply chaoscenter/graphql/server/pkg/authorization/roles.go:14]: Yes it needs to be updated everywhere where these are accessed. Please make sure to validate the changes by running the server locally, before pushing the changes.

[the remark this answers is not in the document]
Comment (RipulHandoo) [reply chaoscenter/graphql/server/pkg/authorization/roles.go:14]: Updated everywhere as required. About to set up the litmus locally for thorough validation.

**Comment by namkyu1999** [comment]
Hey @RipulHandoo , Can you check `build-pipeline / docker-build-graphql-server (pull_request)` ?

**Comment by RipulHandoo** [comment]
> Hey @RipulHandoo , Can you check `build-pipeline / docker-build-graphql-server (pull_request)` ?

@namkyu1999  PTAL!

File: chaoscenter/graphql/server/pkg/authorization/roles.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (RipulHandoo) [reply chaoscenter/graphql/server/pkg/authorization/roles.go]: Hello @Saranya-jena,

I have implemented the changes as per your request. PTAL.

**Comment by RipulHandoo** [comment]
> Hey @RipulHandoo , Can you check `build-pipeline / docker-build-graphql-server (pull_request)` ?

@namkyu1999 PTAL!

File: chaoscenter/graphql/server/pkg/authorization/roles.go [hunk 3]
```diff
@@ -8,154 +8,101 @@ import (
...
-	UpdateProbe:                  {MemberRoleOwnerString, MemberRoleEditorString},
-	GetProbe:                     {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	ListProbes:                   {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
-	DeleteProbe:                  {MemberRoleOwnerString, MemberRoleEditorString},
+	ListInfrastructures:   {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
+	GetInfrastructure:     {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
+	GetManifest:           {MemberRoleOwnerString, MemberRoleEditorString},
+	GetInfraDetails:       {MemberRoleOwnerString, MemberRoleEditorString},
+	ListCharts:            {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
+	ListEnvironment:       {MemberRoleOwnerString, MemberRoleEditorString, MemberRoleViewerString},
```
Comment (namkyu1999) [inline chaoscenter/graphql/server/pkg/authorization/roles.go:90]: same

File: chaoscenter/graphql/server/pkg/authorization/roles.go [hunk 4]
```diff
@@ -8,154 +8,101 @@ import (
...
+	SaveChaosHub:          {MemberRoleOwnerString, MemberRoleEditorString},
+	CreateImageRegistry:   {MemberRoleOwnerString},
+	UpdateImageRegistry:   {MemberRoleOwnerString},
+	DeleteImageRegistry:   {MemberRoleOwnerString},
+	GetGitOpsDetails:      {MemberRoleOwnerString},
+	ListImageRegistry:     {MemberRoleOwnerString},
+	GetImageRegistry:      {MemberRoleOwnerString},
+	CreateEnvironment:     {MemberRoleOwnerString, MemberRoleEditorString},
```
Comment (namkyu1999) [inline chaoscenter/graphql/server/pkg/authorization/roles.go:98]: same

**Review by namkyu1999 — CHANGES_REQUESTED** [review]
Check my comments, please!

File: chaoscenter/graphql/server/graph/chaos_experiment.resolvers.go [hunk 5]
```diff
@@ -165,7 +165,7 @@ func (r *queryResolver) GetExperiment(ctx context.Context, projectID string, exp
 	}
 	logrus.WithFields(logFields).Info("request received to get chaos experiment")
 	err := authorization.ValidateRole(ctx, projectID,
-		authorization.MutationRbacRules[authorization.ListWorkflow],
+		authorization.MutationRbacRules[authorization.ListEnvironment],
```
[the remark this answers is not in the document]
Comment (RipulHandoo) [reply chaoscenter/graphql/server/graph/chaos_experiment.resolvers.go:168]: Yes, there is a typo error. It should be `ListExperiment.` With version 3.0,` WorkFlow` has been changed to `-> Experiment`

File: chaoscenter/graphql/server/pkg/authorization/roles.go (hunk 4 above)
Comment (RipulHandoo) [reply chaoscenter/graphql/server/pkg/authorization/roles.go:98]: The  `CreateEnvironment` should not be changed to `CreateExperiment`. It is a part of both [environment.resolver.go](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/graph/environment.resolvers.go) and [chaos_experiment.resolver.go](https://github.com/litmuschaos/litmus/blob/master/chaoscenter/graphql/server/graph/chaos_experiment.resolvers.go). I've intentionally kept it in the `Chaos-Experiment` section.

File: chaoscenter/graphql/server/graph/chaos_experiment.resolvers.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (RipulHandoo) [reply chaoscenter/graphql/server/graph/chaos_experiment.resolvers.go]: I've made the changes. PTAL.

**Review by namkyu1999 — APPROVED** [review]

**Review by SarthakJain26 — APPROVED** [review]

**Review by Saranya-jena — APPROVED** [review]

**Pull request merged**